src/flint/fq_nmod_mpoly.jl

```diff
@@ -0,0 +1,974 @@
+###############################################################################
+#
+#   fq_nmod_mpoly.jl : Flint multivariate polynomials over fq_nmod
+#
+###############################################################################
+
+export FqNmodMPolyRing, fq_nmod_mpoly
+
+###############################################################################
+#
+#   Data type and parent object methods
+#
+###############################################################################
+
+parent_type(::Type{fq_nmod_mpoly}) = FqNmodMPolyRing
+
+elem_type(::Type{FqNmodMPolyRing}) = fq_nmod_mpoly
+
+elem_type(::FqNmodMPolyRing) = fq_nmod_mpoly
+
+symbols(a::FqNmodMPolyRing) = a.S
+
+parent(a::fq_nmod_mpoly) = a.parent
+
+function check_parent(a::fq_nmod_mpoly, b::fq_nmod_mpoly)
+   parent(a) != parent(b) &&
+      error("Incompatible polynomial rings in polynomial operation")
+end
+
+nvars(a::FqNmodMPolyRing) = a.nvars
+
+base_ring(a::FqNmodMPolyRing) = a.base_ring
+
+base_ring(f::fq_nmod_mpoly) = f.parent.base_ring
+
+function ordering(a::FqNmodMPolyRing)
+   b = a.ord
+#   b = ccall((:fq_nmod_mpoly_ctx_ord, libflint), Cint, (Ref{FqNmodMPolyRing}, ), a)
```

Comment:
can be deleted?

Reply:
probably - it is copy pasta left over from nmod, see also nmod_mpoly.jl. If the struct members do line up, then the simple `b=a.ord` will do. Otherwise, the commented ccall is the proper way.

Reply:
Do the struct members all line up for every version of Julia right now? If so then we can just remove the ccall line. But if this is likely to change then perhaps we are better off with the ccall after all.

Or did I misunderstand what you meant?